src/Messages/Interactions/TSMessage.h

```diff
@@ -60,6 +64,11 @@ typedef NS_ENUM(NSInteger, TSGroupMetaMessage) {
 
 - (BOOL)hasAttachments;
 
+// This message should return TSMessage.receivedAtDate for most messages.
+// For messages received before TSMessage.receivedAtDate was added, this
+// will try to return TSIncomingMessage.receivedAt.
+- (nullable NSDate *)bestReceivedAtDate;
```

Comment:
Okay, I've removed the `receivedAt` property and added a migration. Please confirm that it is correct, since I haven't done a YapDatabase migration before.  

Also, `bestReceivedAtDate` (now called `receiptDateForSorting`) is a method, not a field.